wger/exercises/api/views.py

```diff
@@ -57,12 +60,15 @@ class ExerciseViewSet(viewsets.ModelViewSet):
                      'license',
                      'license_author')
 
-    def pre_save(self, obj):
+    def perform_create(self, serializer):
         '''
-        Set language, author and status
+        Set author and status
         '''
-        obj.language = load_language()
+        language = load_language()
+        obj = serializer.save(language=language)
+        # Todo is it right to call set author after save?
```

Comment:
Yes, I tried to place the logic to other places (model, serializer), but had no luck because of the request. I will try to look into it, maybe there is a better way.